explorer/explorer.go

```diff
@@ -618,26 +621,41 @@ func (exp *explorerUI) Store(blockData *blockdata.BlockData, msgBlock *wire.MsgB
 
 	// Do not run updates if blockchain sync is running.
 	if !exp.AreDBsSyncing() {
-		// Update the charts data after every five blocks or if no charts data
-		// exists yet.
-		if newBlockData.Height%5 == 0 || cacheChartsData.Height() == -1 {
-			// This must be done after storing BlockInfo since that provides the
-			// explorer's best block height, which is used by prePopulateChartsData
-			// to decide if an update is needed.
-			go exp.prePopulateChartsData()
+		// Politeia updates happen hourly thus if every blocks takes an average
+		// of 5 minutes to mine then 12 blocks take approximately 1hr.
+		// https://docs.decred.org/advanced/navigating-politeia-data/#voting-and-comment-data
+		if newBlockData.Height%12 == 0 {
+			// Update the proposal DB. This is run asynchronously since it involves
+			// a query to Politeia (a remote system) and we do not want to block
+			// execution.
+			go func() {
+				err := exp.proposalsSource.CheckProposalsUpdates()
+				if err != nil {
+					log.Error(err)
+				}
+			}()
 		}
 
-		// Update the proposal DB. This is run asynchronously since it involves
-		// a query to Politeia (a remote system) and we do not want to block
-		// execution.
-		if newBlockData.Height%20 == 0 {
+		// Update in every 5 blocks implys that in approximately every 25mins
+		// an update will be queried.
+		if newBlockData.Height%5 == 0 {
+			// Update the Agendas DB. Run this asynchronously to avoid
+			// blocking other processes.
 			go func() {
-				err := exp.proposalsSource.CheckProposalsUpdates()
+				err := exp.agendasSource.CheckAgendasUpdates(exp.client,
```

Comment:
The need to make a `client` field in `explorerUI` just to support this operation is not appealing.  Note that a way around this would be to have the explorerUI constructor take as an input argument a `trigger func()`.  But this just makes it clearer that it's really not the place for this. (BTW, seems like agendaSouce should/could have it's own internal client.)  Anyway,  I think the other option of making and registering another a BlockDataSaver (or two) that only calls the agenda ad proposal DB updates, and does not use the actual block data. Each type, the agenda instance and proposal instance can implement Store for this reason.

Reply:
What if I passed the rpc client as an argument like
 `func NewAgendasDB(dbPath string, client DeploymentSource) (*AgendaDB, error) {`
?
BlockDataSaver interface shall try to pass arguments that may not be that useful in both agendasDB and ProposalDB update.